### DISCUSSION TITLE
Best approach for global functions in Typescript

### DISCUSSION THREAD

**Discussion by unknowntester404** [webdriverio/webdriverio#12673] [Q&A]
Hello,

What will be the best approach to have functions available globally?

Currently, in JS I have global.getElement(selector) { const element = await $(selector)
return element}
but this approach does not work with ts.
What will be the best approach to use getElement function globally in all files from the entire repo something like get myLocator {
const myElement = getElement('#myid#)
}

Thanks

**Comment by erwinheitzman** [comment]
I suggest you extend the browser object using the addCommand command

**Reply by unknowntester404** [reply]
I have driver.addCommand in the file but are not recognized globally as it was when using JS

**Reply by erwinheitzman** [reply]
Are you using typescript now? If so you have to add the types

**Reply by unknowntester404** [reply]
How should the types from tsconfig should look like in order to have the commands available globally. And how should the extended browser should look like? Can’t seem to find anything via google which actually works

**Comment by christian-bromann** [comment]
Another option would be to use [Page Objects](https://webdriver.io/docs/pageobjects) which you can import in all files and allow you to manage selectors and elements.

**Reply by unknowntester404** [reply]
Yeah, was thinking of having a global.d.ts file and have export functions there and import them where needed.

but i still have the case where I want to set a global variable in before hook with a value and overwrite it later on and can’t seem to find a proper way of global vars
